src/gluonts/torch/model/tide/module.py

```diff
@@ -0,0 +1,429 @@
+
+        self.decoder = nn.Sequential(*layers)
+
+    def forward(self, x):
+        return self.decoder(x)
+
+
+class TemporalDecoder(nn.Module):
+    def __init__(
+        self,
+        input_dim: int,
+        hidden_dim: int,
+        output_dim: int,
+        dropout_rate: float,
+        layer_norm: bool,
+    ):
+        super().__init__()
+
+        self.temporal_decoder = ResBlock(
+            input_dim,
+            hidden_dim,
+            output_dim,
+            dropout_rate,
+            layer_norm,
+        )
+
+    def forward(self, x):
+        return self.temporal_decoder(x)
+
+
+class TiDEModel(nn.Module):
+    """
+
+    Parameters
+    ----------
+    context_length
+        Number of time steps prior to prediction time that the model
+        takes as inputs.
+    prediction_length
+        Length of the prediction horizon.
+    num_feat_dynamic_proj
+        Output size of feature projection layer.
+    num_feat_dynamic_real
+        Number of dynamic real features in the data.
+    num_feat_static_real
+        Number of static real features in the data.
+    num_feat_static_cat
+        Number of static categorical features in the data.
+    cardinality
+        Number of values of each categorical feature.
+        This must be set if ``num_feat_static_cat > 0``.
+    embedding_dimension
+        Dimension of the embeddings for categorical features.
+    feat_proj_hidden_dim
+        Size of the feature projection layer.
+    encoder_hidden_dim
+        Size of the dense encoder layer.
+    decoder_hidden_dim
+        Size of the dense decoder layer.
+    temporal_hidden_dim
+        Size of the temporal decoder layer.
+    distr_hidden_dim
+        Size of the distribution projection layer.
+    decoder_output_dim
+        Output size of dense decoder.
+    dropout_rate
+        Dropout regularization parameter.
+    num_layers_encoder
+        Number of layers in dense encoder.
+    num_layers_decoder
+        Number of layers in dense decoder.
+    layer_norm
+        Enable layer normalization or not.
+    distr_output
+        Distribution to use to evaluate observations and sample predictions.
+    scaling
+        Which scaling method to use to scale the target values.
+
+
+    """
+
+    @validated()
+    def __init__(
+        self,
+        context_length: int,
+        prediction_length: int,
+        num_feat_dynamic_real: int,
+        num_feat_dynamic_proj: int,
+        num_feat_static_real: int,
+        num_feat_static_cat: int,
+        cardinality: List[int],
+        embedding_dimension: Optional[List[int]],
+        feat_proj_hidden_dim: int,
+        encoder_hidden_dim: int,
+        decoder_hidden_dim: int,
+        temporal_hidden_dim: int,
+        distr_hidden_dim: int,
+        decoder_output_dim: int,
+        dropout_rate: float,
+        num_layers_encoder: int,
+        num_layers_decoder: int,
+        layer_norm: bool,
+        distr_output: DistributionOutput,
+        scaling: str,
```

Comment:
I think it would be nice to have default values also at the module level, so that this class could be easily used directly. You can set the defaults that are implied by the default estimator config, maybe?

Reply:
Will this lead to some duplications? My understanding is that the estimator, which is also the interface to users,  is responsible for setting these parameters, while the model code in module.py takes whatever it receives from the interface (estimator.py). If a user is also expected to use this model in module.py directly, I totally agree that these default values should also be there. Could you give such a use case?